apis/tempo/v1alpha1/tempostack_types.go

```diff
@@ -170,6 +177,23 @@ type TracingConfigSpec struct {
 	JaegerAgentEndpoint string `json:"jaeger_agent_endpoint,omitempty"`
 }
 
+type DatasourcesConfigSpec struct {
+	// CreateDatasource specifies if a Datasource should be created for Tempo.
+	//
+	// +optional
+	// +kubebuilder:validation:Optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Create CreateDatasource for Tempo"
+	CreateDatasource bool `json:"createServiceMonitors,omitempty"`
+
+	// JaegerAgentEndpoint defines the jaeger endpoint data gets send to.
+	//
+	// +optional
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default:="localhost:8080"
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Tempo-Gateway-Endpoint"
+	TempoGatewayEndpoint string `json:"tempo_gateway_endpoint,omitempty"`
```

Comment:
The operator can figure out the endpoint. It's either query-frontend or the gateway (if enabled), for example
```
http://tempo-simplest-query-frontend.$namespace.svc:3200
```
The service name is generated here: https://github.com/grafana/tempo-operator/blob/a701112a9c1b3557b4c3b3d828c84629b33ca597/internal/manifests/queryfrontend/query_frontend.go#L342
The namespace of the service is in `tempo.Namespace` (where tempo is the `TempoStack` instance).

Reply:
I was unsure if I could use it or not. Will take that into consideration then, thanks.

Reply:
I have update the URL logic, could you please validate it?